src/Controller.res

```diff
@@ -5,20 +5,33 @@ module OnChangeArg: {
 
   type rec t
 
+  type kind = Event(ReactEvent.Form.t) | Value(Js.Json.t)
+
   @ocaml.doc("Takes a `ReactEvent.Form.t` and return an `onChange` argument.
 Useful for dealing with native form inputs.")
   let event: ReactEvent.Form.t => t
 
   @ocaml.doc("Takes a `Js.Json.t` and return an `onChange` argument.
 This function allows you to set the input value to any arbitrary value.")
   let value: Js.Json.t => t
+
+  let classify: t => kind
 } = {
   @unboxed
   type rec t = Any('a): t
 
+  type kind = Event(ReactEvent.Form.t) | Value(Js.Json.t)
+
   let event = eventHandler => Any(eventHandler)
 
   let value = valueHandler => Any(valueHandler)
+
+  let classify = (Any(unknown)) =>
+    unknown->Js.typeof == "object" &&
+    unknown->Js.Nullable.return->Js.Nullable.isNullable->not &&
+    Obj.magic(unknown)["_reactName"] == "onChange"
+      ? Event(Obj.magic(unknown))
+      : Value(Obj.magic(unknown))
```

Comment:
just to confirm this one is just for checking what kind of type of Argument that passed via the onClick function in rescript-hook-form right @gaku-sei .

Reply:
Exactly. To be honest it's not perfect and you can "fool" it, but it's rather unlikely to happen 😕